doc/persistence.rst

```diff
@@ -0,0 +1,103 @@
+..
+    For doctests:
+
+    >>> from joblib.testing import warnings_to_stdout
+    >>> warnings_to_stdout()
+
+.. _persistence:
+
+===========
+Persistence
+===========
+
+.. currentmodule:: joblib.numpy_pickle
+
+Usecase
+=======
+
+:func:`joblib.dump` and :func:`joblib.load` provide a replacement for
+pickle to work efficiently on Python objects containing large data, in
+particular large numpy arrays.
+
+A simple example
+================
+
+  First we create a temporary directory:
+    >>> from tempfile import mkdtemp
+    >>> savedir = mkdtemp()
+    >>> import os
+    >>> filename = os.path.join(savedir, 'test.pkl')
+
+  Then we create an object to be persisted:
+    >>> import numpy as np
+    >>> to_persist = [('a', [1, 2, 3]), ('b', np.arange(10))]
+
+  which we save into `savedir`:
+    >>> import joblib
+    >>> joblib.dump(to_persist, filename)  # doctest: +ELLIPSIS
+    ['...test.pkl', '...test.pkl_01.npy']
+
+  We can then load the object from the file:
+    >>> joblib.load(filename)
+    [('a', [1, 2, 3]), ('b', array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]))]
```

Comment:
Maybe add a note to make it explicit that contrary to default `pickle` joblib tends to use several files to store the binary content of internal numpy arrays separately from the main object datastructure. Those auxiliary files needs to be kept around to be able to load the object in the main file later.